#!/usr/bin/python
PACKAGE = "teraranger_evo_64px"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("depth_image_invert", bool_t, 0, "Invert depth Image", False)
gen.add("depth_image_interpolate", bool_t, 0, "Interpolate depth Image", False)
gen.add("min_distance_mm", double_t, 1, "Min distance allowed", 100.0, 0.0, 5000.0)
gen.add("max_distance_mm", double_t, 2, "Max distance allowed", 5000.0, 0.0, 5000.0)


mode_enum = gen.enum([gen.const("Close_Range", int_t, 0, "Allows ranging from 10cm to 5m"),
                               gen.const("Fast", int_t, 1, "Allows ranging from 1m to 5m with faster update rate")],
                              "Set the sensor mode")

gen.add("Mode", int_t, 3, "Set the sensor mode", 0, 0, 1, edit_method=mode_enum)

exit(gen.generate(PACKAGE, "teraranger_evo_64px", "Evo64px"))
